test/dialects/dialects_test.py

```diff
@@ -22,57 +24,111 @@
 )
 
 
-@pytest.mark.parametrize("dialect,file", parse_success_examples)
-def test__dialect__base_file_parse(dialect, file):
-    """For given test examples, check successful parsing."""
-    raw = load_file(dialect, file)
+def _lex_and_parse(config_overrides: str, raw: str) -> Optional[BaseSegment]:
+    """Performs a Lex and Parse, with cachable inputs within fixture."""
     # Load the right dialect
-    config = FluffConfig(overrides=dict(dialect=dialect))
+    config = FluffConfig(overrides=json.loads(config_overrides))
     tokens, lex_vs = Lexer(config=config).lex(raw)
     # From just the initial parse, check we're all there
     assert "".join(token.raw for token in tokens) == raw
     # Check we don't have lexing issues
     assert not lex_vs
-
     # Do the parse WITHOUT lots of logging
     # The logs get too long here to be useful. We should use
     # specific segment tests if we want to debug logs.
-    if raw:
-        parsed = Parser(config=config).parse(tokens)
-        print(f"Post-parse structure: {parsed.to_tuple(show_raw=True)}")
-        print(f"Post-parse structure: {parsed.stringify()}")
-        # Check we're all there.
-        assert parsed.raw == raw
-        # Check that there's nothing unparsable
-        typs = parsed.type_set()
-        assert "unparsable" not in typs
+    if not raw:
+        return None
+
+    return Parser(config=config).parse(tokens)
+
+
+def lex_and_parse(config_overrides: Dict[str, Any], raw: str) -> Optional[BaseSegment]:
+    """Sets up cache parsable inputs from normal inpurts."""
+    cachable_str = json.dumps(config_overrides, sort_keys=True)
+    return _lex_and_parse(cachable_str, raw)
+
+
+@pytest.mark.parametrize("dialect,file", parse_success_examples)
+def test__dialect__base_file_parse(dialect, file):
+    """For given test examples, check successful parsing."""
+    raw = load_file(dialect, file)
+    config_overides = dict(dialect=dialect)
+    # Use the helper function to avoid parsing twice
+    parsed: Optional[BaseSegment] = lex_and_parse(config_overides, raw)
+    if not parsed:
+        return
+
+    print(f"Post-parse structure: {parsed.to_tuple(show_raw=True)}")
+    print(f"Post-parse structure: {parsed.stringify()}")
+    # Check we're all there.
+    assert parsed.raw == raw
+    # Check that there's nothing unparsable
+    typs = parsed.type_set()
+    assert "unparsable" not in typs
+
+
+@pytest.mark.integration_test
+@pytest.mark.parametrize("dialect,file", parse_success_examples)
+def test__dialect__base_broad_fix(dialect, file, raise_critical_errors_after_fix):
+    """Run a full fix with all rules, in search of critical errors."""
+    raw = load_file(dialect, file)
+    config_overides = dict(dialect=dialect)
+    # Lean on the cached result of the above test if possible
+    parsed: Optional[BaseSegment] = lex_and_parse(config_overides, raw)
```

Comment:
Do any of the test files use Jinja? I'm confused that both tests call `lex_and_parse()` with the raw file contents, but this test calls the templater, but only _after_ parsing. It seems like this couldn't possibly work if the files use templating.

Did you consider calling a higher level `Linter` functions directly, similar to the test functions in this file? https://github.com/sqlfluff/sqlfluff/blob/main/src/sqlfluff/testing/rules.py

Reply:
None of our dialect fixtures use Jinja.

Reply:
Yup thats the way I started out, but then went about caching the Parse as it was expensive and slow.

Reply:
I made this simpler